torchbenchmark/util/framework/detectron2/model_factory.py

```diff
@@ -90,6 +90,8 @@ def __init__(self, variant, test, device, jit=False, batch_size=None, extra_args
             args.fcos_use_bn = True
 
         cfg = setup(args)
+        if getattr(cfg, "MODEL", None) and cfg.MODEL.DEVICE != self.device:
+            cfg.MODEL.DEVICE == self.device
```

Comment:
Should it be `cfg.MODEL.DEVICE = self.device`?
```suggestion
            cfg.MODEL.DEVICE = self.device
```

Reply:
keen eyes! This is fixed